contracts/FeralfileArtworkV4_2.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {Nonces} from "./Nonces.sol";
+import {FeralfileExhibitionV4_1} from "./FeralfileArtworkV4_1.sol";
+import {IFeralfileVaultV2} from "./IFeralfileVaultV2.sol";
+import {FeralfileSaleDataV2} from "./FeralfileSaleDataV2.sol";
+
+contract FeralfileExhibitionV4_2 is
+    FeralfileExhibitionV4_1,
+    FeralfileSaleDataV2,
+    Nonces
+{
+    error TokenIDNotFound();
+    error FunctionNotSupported();
+    error SaleNotStarted();
+    error InvalidPaymentAmount();
+    error TotalBpsOver();
+    error InvalidAddress();
+
+    // vault contract instance
+    IFeralfileVaultV2 public vaultV2;
```

Comment:
Same [here](https://github.com/bitmark-inc/feralfile-exhibition-smart-contract/pull/42/files#r1634594351). 
There is no another vault v1 in this contract so it doesn't make us confused, so it should be named as `vault` and `setVault` is clear enough and remove the confusion.